openassessment/fileupload/backends/gcs.py

```diff
@@ -0,0 +1,73 @@
+"""GCS Bucket File Upload Backend."""
+
+from google.cloud import storage
+import logging
+
+from ..exceptions import FileUploadInternalError
+from .base import BaseBackend
+
+log = logging.getLogger("openassessment.fileupload.api")  # pylint: disable=invalid-name
+
+
+class GCSBackend(BaseBackend):
+    """ GCS Bucket File Upload Backend. """
+
+    def get_upload_url(self, key, content_type):
+        """Get a signed URL for uploading a file to GCS"""
+        bucket_name, key_name = self._retrieve_parameters(key)
+        try:
+            client = _connect_to_gcs()
+            bucket = client.bucket(bucket_name)
+            blob = bucket.blob(key_name)
+            url = blob.generate_signed_url(
+                version="v4",
+                expiration=self.UPLOAD_URL_TIMEOUT,
+                method="PUT",
+                content_type=content_type,
+            )
+            return url
+        except Exception as ex:
+            log.exception("An internal exception occurred while generating an upload URL.")
+            raise FileUploadInternalError(ex) from ex
```

Comment:
I think you can use the message in this way:
```suggestion
            msg = "An internal exception occurred while generating an upload URL."
            log.exception(msg)
            raise FileUploadInternalError(msg) from ex
```

the `from ex` part will also give you information from the previous exception (`ex`) and its message.